Random forest

xiansch opened the pull request:
Hi, I cleaned the random forest and merged the screening datasets which are currently loaded from npz as generated by the last kernel of screening_data notebook.  Could also be saved as pickle but in my case pandas complained about the dimensionality of the fingerprint array.  Actually the random forest doesn't take very long to run at all if no negative training examples are used.  Top hits are saved in the output from the random_forest_clean notebook (second to last kernel).  
The screening datasets are combined from ReFrame, Broad, and Drugbank and might need to be cleaned (Drugbank sdf spat out some errors which I left in screening_data.ipynb).  Nafamostat appears to be included in all of the databases and probably more than once (different configurations or smiles code parsing issue?) with slight different values depending on the database source, but always in top hits.

Commit:
random forest

Commit:
random forest with negative examples and regression

Commit:
random forest with dark data

Commit:
Merge branch 'master' into random-forest

Commit:
random forest with merged datasets

Commit:
random forest with screening data

Commit:
run regression without dark data for now

Commit:
merge screening data and clean random forest

Commit:
merge screening data and clean random forest

Commit:
fixed an error in standardization

Commit:
fixed an error in standardization

dklink committed:
Merge branch 'master' into random-forest

dklink committed:
cleared output

dklink commented on dumps/random_forest_withdark.ipynb:
```diff
@@ -0,0 +1,430 @@
+{
```
<p>Looks like this is an old version of notebooks/random_forest_withdark.ipynb.  Plz delete</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/srensi/tmprss2/pull/20/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='0'/>

dklink commented on notebooks/random_forest_clean.ipynb:
```diff
@@ -0,0 +1,442 @@
+{
```
<p>rg_scaled not defined? Stuff like this should get tidied up before we merge to master.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/srensi/tmprss2/pull/20/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='10'/>

dklink replied:
<p>make sure to add ../dumps/screening_fp.npz to git, it doesn't exist before running the screening notebook</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/srensi/tmprss2/pull/20/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='15'/>

dklink replied:
<p>rg_all2 not defined.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/srensi/tmprss2/pull/20/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='17'/>

dklink commented on notebooks/random_forest_withdark.ipynb:
```diff
@@ -0,0 +1,342 @@
+{
```
<p>drop_duplicates operates on and returns a copy of the dataframe, so line 29 does nothing.  You either want</p><p>features = features.drop_duplicates, or features.drop_duplicates(..., inplace=True)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/srensi/tmprss2/pull/20/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='0'/>

dklink replied:
<p>This errors?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/srensi/tmprss2/pull/20/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

dklink commented:
Aborting review, there's a little bit more cleanup that needs to happen.  Shanna, if you could just go through your notebooks and remove any leftover code snippets, commented out blocks, just general cleanup stuff, and then make sure everything cleanly runs without errors, that'll make the review process a lot easier.  Also, adding a bit of documentation could be helpful :)

dklink committed:
updated data loading to reflect current locations/names

dklink commented:
One more thing you'll want to do though, Shanna: Kaan and I deprecated all the `<target>_morgan_fingerprints.npz` files, because the fingerprints are now included in the processed data files (e.g. all the .pkl files in processed_data/).  Access the fingerprints like this:
```
>>> import pandas as pd
>>> import numpy as np
>>> ds = pd.read_pickle('processed_data/combined_dataset.pkl')
>>> fp = np.stack(ds.morgan_fingerprint)
>>> fp.shape
(895, 2048)
```

The deprecated files are still there, in `dumps/deprecated/`, and I've updated your code to read them from there, so your code still runs, BUT if/when we make changes upstream to the data processing upstream, the deprecated files WILL NOT be updated, while all the files in processed_data will be updated.  Ideally we will delete everything in `dumps/deprecated/` in the near future.

Commit:
remove old files

Commit:
delete old files

Commit:
remove old notebooks

Commit:
added markdown to notebook

xiansch commented:
deleted some of the old notebooks, the only one up now should be "random_forest_clean"?  Also added markdown to that notebook and changed all data i/o to pickle files, including the screening data that was just added.

dklink committed:
Merge branch 'master' into random-forest

dklink committed:
load dcm from pickle, violinplots

dklink committed:
remove dumps/random_forest_withdark.ipynb (presumably shanna forgot to delete this)
remove notebooks/screening_data.ipynb (already a processor in master)
remove screening data from dumps (already in data/screening_data)

dklink commented on notebooks/random_forest_clean.ipynb:
```diff
@@ -0,0 +1,1507 @@
+{
```
<p>this random shuffling is not controlled by a hard-coded seed; thus not reproducible.  Every time the notebook is re-run, there is a different train and test split, and thus different validation results.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/srensi/tmprss2/pull/20/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

xiansch replied:
<p>added random.seed() command</p>

dklink commented on notebooks/random_forest_clean.ipynb:
```diff
@@ -0,0 +1,1507 @@
+{
```
<p>If you like the below visualization better, delete these two scatter plots</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/srensi/tmprss2/pull/20/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='22'/>

xiansch replied:
<p>oh nice! I like the violin plot, let's keep that one</p>

dklink commented on notebooks/random_forest_clean.ipynb:
```diff
@@ -0,0 +1,1507 @@
+{
```
<p>if you like the below visualization/table better, just delete this cell</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/srensi/tmprss2/pull/20/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='31'/>

dklink commented:
@xiansch in the random forest notebook, I did the last i/o change, which is getting DCM from processed_data/DarkChemicalMatter_processed.pkl.gz.  I also added some violinplots in two places which might be clearer than your scatter plots.  Delete the old plots if you agree!

Otherwise, I deleted every other file associated with this PR, because they were all outdated :)  nice clean PR now!

Excellent work, exciting results!  And, the notebook is super clean, as the reviewer I hugely appreciate your effort to streamline things.  Who would have thought it could be streamlined so much... 5 notebooks to 1!  Cheers

Commit:
more notebook tidying and pretty visuals

dklink commented:
looks great, merging!
